piccolo/tablestorage.py

```diff
@@ -0,0 +1,218 @@
+
+        :return:
+            None
+        """
+        dict.clear(self.tables)
+        self._schema_tables.clear()
+
+    async def get_table(self, tablename: str) -> t.Optional[t.Type[Table]]:
+        """
+        Returns the Table object if exists. if table is not present in the
+        TableStorage, it will try to reflect it.
+        :param tablename:
+            Name of the table, schema name included.
+            If schema is public, It's not necessary.
+            for example: "public.manager" or "manager",
+            "test_schema.test_table"
+        :return:
+            Table | None
+        """
+        table_class = self.tables.get(tablename)
+        if table_class is None:
+            tableNameDetail = self._get_schema_and_table_name(tablename)
+            await self.reflect(
+                schema_name=tableNameDetail.schema,
+                include=[tableNameDetail.name],
+            )
+            table_class = self.tables.get(tablename)
+        return table_class
+
+    async def _add_table(self, schema_name: str, table: t.Type[Table]) -> None:
+        if isinstance(table, TableMetaclass):
+            table_name = self._get_table_name(
+                table._meta.tablename, schema_name
+            )
+            self.tables._insert_item(table_name, table)
+            self._add_to_schema_tables(
+                schema_name=schema_name, table_name=table._meta.tablename
+            )
+
+    def _add_to_schema_tables(self, schema_name: str, table_name: str) -> None:
+        """
+        We keep record of schemas and their tables for easy use.
+        This method adds a table to its schema.
+        """
+        schema_tables = self._schema_tables.get(schema_name)
+        if schema_tables is None:
+            self._schema_tables[schema_name] = []
+        else:
+            self._schema_tables[schema_name].append(table_name)
+
+    @staticmethod
+    def _get_table_name(name: str, schema: str):
+        if schema == "public":
+            return name
+        else:
+            return schema + "." + name
+
+    def __repr__(self):
+        return f"{self.tables}"
+
+    @staticmethod
+    def _get_schema_and_table_name(tablename: str) -> TableNameDetail:
+        """
+        Extract schema name from tablename attribute.
+        :param tablename:
+            Full name of the table
+        :return:
+            Returns the name of the schema and the table.
+        """
+        # Bad naming, I know :))
+        # And maybe not the best idea for handling schema names.
+        # Could use your suggestion.
+        tablenamelist = tablename.split(".")
+        if len(tablenamelist) == 2:
+            return TableNameDetail(
+                name=tablenamelist[1], schema=tablenamelist[0]
+            )
+
+        elif len(tablenamelist) == 1:
+            return TableNameDetail(name=tablenamelist[0], schema="public")
+        else:
+            raise ValueError("Couldn't find schema name.")
```

Comment:
@dantownsend What do you think about this part?
can we handle this more gracefully?

Reply:
It looks OK to me. I'd have have to test it on a proper schema to be sure, but in theory can't see any issues.

Reply:
I'll write a test for this to be more sure.
There is only one case that comes to my mind. If there is a `.` in the name of a table, this will fail.

Reply:
I was 99% sure that `.` wasn't allowed in table names, but just tried it, and it seems like it's possible.

Reply:
The same seems to be true of schema names, which complicates matters too.

Reply:
Yes, I tried them the other day and that worries me.
We can raise an informative exception when the pattern is different and put `schema_name` as an optional parameter for this case.
This probably won't fit well with the rest of the API though.
A better approach could be to implement schema support directly into the whole API. But I think there will be some problems regarding backward compatibility that needs to be handled to prevent breaking changes.
Was there any technical reason for not separating table_name and schema_name from each other when you started working on piccolo?

Reply:
When I'm building a web app, I typically just use the default schema, so it never really occurred to me to add support for different schemas.

It might be possible to retrofit it to Piccolo without any breakage with a `schema` argument in `Table`.

We can probably assume that in practice nobody puts a `.` character in either their table names or schema names.

Reply:
Yeah, I think that would be a pretty safe bet, but still a bet.
So first we try to use `schema` if specified, else we fall back to splitting `tablename`, right?

Reply:
Yeah, should work.

Reply:
Okay. So I'll leave this as it is and work on this matter on another PR.

Reply:
Cool, makes sense - thanks